utilities/blob_db/blob_compaction_filter.cc

```diff
@@ -180,7 +180,7 @@ bool BlobIndexCompactionFilterBase::OpenNewBlobFileIfNeeded() const {
     ROCKS_LOG_ERROR(blob_db_impl->db_options_.info_log,
                     "Error opening new blob file during GC, status: %s",
                     s.ToString().c_str());
-
+    blob_file_.reset();
```

Comment:
Could you also reset `writer_`? It is possible for `CreateBlobFileAndWriter` to fail after the `Writer` has been created (namely in `WriteHeader`).

Reply:
> Could you also reset `writer_`? It is possible for `CreateBlobFileAndWriter` to fail after the `Writer` has been created (namely in `WriteHeader`).

Make sense, done.